tests/projections/test_projection_specifications.py

```diff
@@ -349,6 +350,30 @@ def test_formats_for_gating_specification_of_input_and_output_ports(self):
         # assert 'Primary OutputPort of ControlMechanism-1 (ControlSignal-0) ' \
         #        'cannot be used as a sender of a Projection to OutputPort of T2' in error_text.value.args[0]
 
+    def test_no_warning_when_matrix_specified(self):
+
+        warning_fired = False
+        c = pnl.Composition()
+        m0 = pnl.ProcessingMechanism(
+            default_variable=[0, 0, 0, 0]
+        )
+        p0 = pnl.MappingProjection(
+            matrix=[[0, 0, 0, 0],
+                    [0, 0, 0, 0],
+                    [0, 0, 0, 0],
+                    [0, 0, 0, 0]]
+        )
+        m1 = pnl.TransferMechanism(
+            default_variable=[0, 0, 0, 0]
+        )
+        with warnings.catch_warnings():
```

Comment:
this can just use `@pytest.mark.filterwarnigns("error:MyWarning")` for the test, or `pytest.warns()` to catch specific warning on a specific line instead of opencoding custom solution.

Reply:
Will update. Thanks for the suggestion.